src/dataSources/DataSourceOrigin.js

```diff
@@ -170,6 +170,47 @@ var DataSourceOrigin = DataSourceBase.extend('DataSourceOrigin',  {
         return result;
     },
 
+    /**
+     * @summary Find, replace, or update a row by it's index.
+     * @param {number} index - Row index that is being accessed
+     * @param {object|null|undefined} [replacement] - One of:
+     * * _omitted_ - Ignored.
+     * * _object_ - Replacement for the data row if found.
+     * * `null` - Flag to delete the data row if found. The found data row is nonetheless returned.
+     * @returns {object|number|undefined} One of:
+     * * `undefined` - data row not found
+     * * _object_ - found data row object (will have been deleted if `replacement` was `null`)
+     * @todo Use a binary search (rather than `Array..find`) when column is known to be indexed (sorted).
+     * @memberOf DataSourceOrigin#
+     */
+    findRowByIndex: function findRow(index, replacement) {
+        var result;
+
+        if (arguments.length < 1) {
+            throw 'Expected at least 1 argument but found ' + arguments.length + '.';
+        }
+
+        if (typeof index !== 'number') {
+            throw 'Expected at index to be a number but got ' + index + '.';
+        }
+        result = this.data.find(function(row, idx) {
```

Comment:
No! Use `getDataIndex` here. Explanation: We need to call getDataIndex to transpose all the indexes. In other words, if the filtered & sorted dataset has 10 rows left out of 1000 and we want to delete row the 7th row, calling `getDataIndex(7-1)` will return the underlying index of that row in `source.data`. Get it?